src/app/shared/components/workshop-modal/workshop-modal.component.ts

```diff
@@ -0,0 +1,25 @@
+import {Component, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA} from '@angular/material/dialog';
+import {Login} from '../../store/registration.actions';
+import {Store} from '@ngxs/store';
+
+interface DialogData {
+  title?: string;
+  message?: string;
+  buttonLabel?: string;
+}
+
+@Component({
+  selector: 'app-workshop-modal',
+  templateUrl: './workshop-modal.component.html',
+  styleUrls: ['./workshop-modal.component.scss']
+})
+
+export class WorkshopModalComponent {
```

Comment:
I believe it is not teh best naming. Obviously, this functionality is not related to the workshop at all, it can be applied everywhere for unregistered user. So, consider better naming

Reply:
Sure, will change the naming. Thank you